### PR TITLE
Change the default cilium configMap value

### PR DESCRIPTION
**Pull request by manuelbuil**
If we want L7 Cilium Network Policies to use the sidecar proxy when
istio is present, sidecar-istio-proxy-image: must point to istio-proxy
instead of cilium/istio_proxy because our proxy image name is
istio-proxy

https://github.com/cilium/cilium/blob/master/pkg/k8s/labels.go#L71-L78
https://github.com/cilium/cilium/blob/master/pkg/k8s/labels.go#L46

Signed-off-by: Manuel Buil <mbuil@suse.com>

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Review by mmnelemane — DISMISSED** [review]

File: internal/pkg/skuba/cni/cilium.go [hunk 1]
```diff
@@ -226,6 +226,7 @@ func CreateOrUpdateCiliumConfigMap(client clientset.Interface, ciliumVersion str
 			"enable-ipv6":              "false",
 			"identity-allocation-mode": "crd",
 			"preallocate-bpf-maps":     "false",
+			"sidecar-istio-proxy-image": "istio-proxy",
```
Comment (mrostecki) [inline internal/pkg/skuba/cni/cilium.go:229]: Does it work for you? I thought that has to be the full image name, including the repository if it's different than `docker.io`. From my understanding, this is going to be evaluated as `docker.io/istio-proxy`.

I think we should rather use something like `registry.suse.de/caasp/.../istio-proxy`, or `registry.suse.com`, depending if we run released skuba or not.

For generating the image names, you can use `GetGenericImage` function, it's used for example here:

https://github.com/SUSE/skuba/blob/98f67addb5bb7c2df56d511681b5115f5b3a9884/internal/pkg/skuba/addons/cilium.go#L41

**Review by mrostecki — CHANGES_REQUESTED** [review]

**Review by mmnelemane — COMMENTED** [review]
Shouldn't this name be `cilium/istio_proxy` as referred by https://github.com/cilium/cilium/blob/master/pkg/k8s/labels.go#L46

File: internal/pkg/skuba/cni/cilium.go (hunk 1 above)
Comment (manuelbuil) [reply internal/pkg/skuba/cni/cilium.go:229]: Yes, I had the same doubt but it works. I checked the function and it looks for `any match of the regular expression`
https://golang.org/pkg/regexp/#Regexp.MatchString

I created a dummy program to verify it:
```
package main

import (
	"fmt"
	"regexp"
)

func main() {
	matching := "istio-proxy"
	myString := "registry.suse.de/devel/caasp/4.5/containers/containers/caasp/v4.5/istio-proxyv2:1.5.9"
	myRegexp := regexp.MustCompile(matching)
	if !myRegexp.MatchString(myString) {
	    fmt.Println("No matching")
	} else {
	    fmt.Println("Matching")
	}
}
```
And returns `Matching`:
```
kosmas@kosmas:~> go run /tmp/throw.go
Matching
```

**Comment by manuelbuil** [comment]
> Shouldn't this name be `cilium/istio_proxy` as referred by https://github.com/cilium/cilium/blob/master/pkg/k8s/labels.go#L46

That's the default value if nothing is provided as a flag of cilium-agent or in the ConfigMap. By changing the configMap, we overwrite that value

File: internal/pkg/skuba/cni/cilium.go (hunk 1 above)
Comment (mrostecki) [reply internal/pkg/skuba/cni/cilium.go:229]: OK, I didn't know that `sidecar-istio-proxy-image` is just a regexp. I though we require the full URI.

Comment (mrostecki) [reply internal/pkg/skuba/cni/cilium.go:229]: But there is another issue with that line of code. You have wrong code formatting here. Run `make lint` locally to see it. ;) You need put all `=` characters on the same level.

Look for some ´gofmt` plugin in your text editor or IDE.

Comment (manuelbuil) [reply internal/pkg/skuba/cni/cilium.go:229]: Check the newest version, that should be fixed (and an error in the unit tests)

**Review by mrostecki — APPROVED** [review]

**Review by mmnelemane — APPROVED** [review]

**Comment by ereslibre** [comment]
Q: without bumping the manifest version, `skuba addon upgrade apply` will not migrate an existing configMap with this change only, is that fine?

**Comment by mmnelemane** [comment]
> Q: without bumping the manifest version, `skuba addon upgrade apply` will not migrate an existing configMap with this change only, is that fine?

Istio has not been released in any stable version of CaaSP so far. So ideally, this will still be the initial version. Should we still be concerned about upgrade ?

**Comment by ereslibre** [comment]
> Istio has not been released in any stable version of CaaSP so far. So ideally, this will still be the initial version. Should we still be concerned about upgrade ?

I see, not necessarily, but when the time comes we will have to remember to bump the manifest version so it gets applied on upgrades -- there's room to forget about it. If we don't forget it shouldn't be a problem though.

**Pull request merged**